scripts/check-docs/src/main.rs

```diff
@@ -0,0 +1,120 @@
+extern crate core;
+
+use std::collections::{HashMap, HashSet};
+use std::env;
+use std::path::Path;
+
+fn main() {
+    let mut stack: Vec<&str> = vec![];
+    let mut valid_anchors: HashMap<&str, HashSet<&str>> = HashMap::new();
+
+    let rg_project = std::process::Command::new("grep")
+        .arg("-I")
+        .arg("-H")
+        .arg("-R")
+        .arg("--exclude-dir=scripts")
+        .arg("ANCHOR")
+        .arg(".")
+        .output()
+        .expect("failed rg command");
+
+    let output_to_string =
+        String::from_utf8(rg_project.stdout).expect("failed to parse command output");
+
+    let split = output_to_string.split('\n');
+
+    let vec_of_anchors = split
+        .into_iter()
+        .filter(|s| !s.is_empty())
+        .map(|g| g.replace("./", "").replace(' ', "").replace("//", ""))
+        .collect::<Vec<_>>();
+
+    for i in &vec_of_anchors {
+        let parts = i.split(':').collect::<Vec<_>>();
+
+        if *parts.get(1).expect("error in parsing vec_of_anchors") == "ANCHOR" {
+            stack.push(*parts.get(2).expect("error"));
+        } else if *parts.get(1).expect("error in parsing vec_of_anchors") == "ANCHOR_END" {
+            if stack.is_empty() {
+                panic!(
+                    "ANCHOR of \"{}\" cannot be found",
+                    *parts.get(2).expect("error")
+                );
+            }
+
+            if stack.pop().unwrap() != *parts.get(2).expect("error") {
+                panic!(
+                    "ANCHOR or ANCHOR_END of \"{}\" is wrong",
+                    *parts.get(2).expect("error")
+                );
+            }
+        } else {
+            panic!(
+                "Invalid syntax for ANCHOR or ANCHOR_END\"{}\"",
+                *parts.get(1).expect("error")
+            );
+        }
+
+        match valid_anchors.contains_key(*parts.get(0).unwrap()) {
+            true => {
+                let update_hash_set = valid_anchors.get_mut(*parts.get(0).unwrap()).unwrap();
+                update_hash_set.insert(*parts.get(2).unwrap());
+            }
+            false => {
+                let mut new_hash_set = HashSet::new();
+                new_hash_set.insert(*parts.get(2).unwrap());
+                valid_anchors.insert(*parts.get(0).unwrap(), new_hash_set);
+            }
+        }
+    }
+
+    let rg_docs = std::process::Command::new("grep")
```

Comment:
Same here for `rg`/`grep` and `arg/args`.

Reply:
I commented it before pushing :D